### DISCUSSION TITLE
[FR] Server to Client packet for displaying AABB selections on the client

### DISCUSSION THREAD

**Discussion by inxomnyaa** [Mojang/minecraft-editor#74] [Feedback]
Would it be possible to get server->client packet that allows a selection to be rendered? I am currently using structure blocks to create outlines in my world editor, but an official way sounds a lot better. Already haven't gotten the Debug Voxels in release clients despite being useful, so at least these new selections would be nice to have.
![image](https://user-images.githubusercontent.com/8733998/224493009-c9fed424-6b6c-4c4b-8d1e-0f3132dd2c18.png)
[![image](https://user-images.githubusercontent.com/8733998/224493153-e295f9ab-d3fb-4fb7-8140-aa60fa23c887.png)https://youtu.be/xntCNHnwl18?t=19](https://youtu.be/xntCNHnwl18?t=19)

**Comment by dacowan** [comment]
If this is in your own extension, then it's currently possible.  Create a new selection object using the context selection manager and set its visibility property. This will result in the selection object being synchronized between client and server.  Just don't do a ton of them because they're not cheap in terms of network bandwidth when they're being adjusted.

**Reply by dacowan** [reply]
> If this is in your own extension, then it's currently possible. Create a new selection object using the context selection manager and set its visibility property. This will result in the selection object being synchronized between client and server. Just don't do a ton of them because they're not cheap in terms of network bandwidth when they're being adjusted.

Oh - and the usual caveat.  If it doesn't work, you're on your own until 0.4 comes out, and it's all subject to change right now - so don't rely on it and be ready to re-write everything again.

**Comment by Xd-pro** [comment]
Or better yet, a way to show a selection via the script API even if you're not in an editor extension

**Reply by inxomnyaa** [reply]
This is what i meant, i guess mentioning it slipped my mind.
My goal is to have "seamless" WorldEdit support on thirdparty servers, where seamless means you can switch between editor mode and my WorldEdit implementation seamlessly, allowing for both visual and in-game selections to work. Selections of my WorldEdit may also be set by other plugins (server plugins, not editor mode extensions).

**Comment by minecraftchloe** [comment]
@inxomnyaa - Curious what your thoughts are with the /server-editor module we just released! [aka.ms/EditorAPI](https://aka.ms/EditorAPI).

**Reply by inxomnyaa** [reply]
what exactly am i looking for there? I don't spot anything related to this feedback

**Reply by minecraftchloe** [reply]
Ah! Sorry - whether or not anything in the API as we released it matches your specific use case. My guess is no, but we want to know if you have suggestions for specific areas of the server-editor module we can investigate improving to support you.
